icaruscode/CRT/CRTTPCTruthEffMCKalman_module.cc

```diff
@@ -0,0 +1,1721 @@
+
+	int tempmin = 4097; int tempmax = -1;
+
+	std::cout << "Found " << (int)hits.size() << " hits to search\n";
+
+	for(int i=0; i<(int)hits.size(); i++){
+
+		int tickhold1 =(int)hits[i]->StartTick(); 
+		if(tickhold1>tempmax) tempmax=tickhold1;
+		if(tickhold1<tempmin) tempmin=tickhold1;
+
+		int tickhold2 =(int)hits[i]->EndTick(); 
+		if(tickhold2>tempmax) tempmax=tickhold2;
+		if(tickhold2<tempmin) tempmin=tickhold2;
+
+//		std::cout << "StartTick: " << tickhold1 << "\tEndTick: " << tickhold2 << std::endl;
+
+	}//end loop over hits
+
+	mintick = tempmin; maxtick = tempmax;
+
+}//end definition of void icarus::CRTTPCTruthEffMCKalman::getTickRange(recob::Track trk, double &mintick, double &maxtick)
+
+void icarus::CRTTPCTruthEffMCKalman::getCatCrossXYZ(recob::Track trk, double &my_x, double &my_y, double &my_z){
+
+	size_t ntrk = trk.NPoints();
+	std::vector<double> x, y, z, x_left_diff, x_right_diff;
+//	std::pair<double,double> cathode_yz;
+	double left_dist_min = DBL_MAX; int left_dist_min_pos = INT_MAX;
+	double right_dist_min = DBL_MAX; int right_dist_min_pos = INT_MAX;
+	double xsum=0, xavg;
+
+	int numelmnts = 0;
+
+	//Begin by extracting coordinates into arrays
+	for(size_t i=0; i<ntrk; i++){
+		geo::Point_t thispt = trk.LocationAtPoint((int)i);
+		x.push_back(thispt.X()); y.push_back(thispt.Y()); z.push_back(thispt.Z()); numelmnts++;
+		if(std::abs(x.back())<210.215) {
+			x_left_diff.push_back(std::abs(std::abs(x[i]) - 210.25));
+			if(x_left_diff.back()<=left_dist_min) { left_dist_min=x_left_diff.back(); left_dist_min_pos = i; }
+		}//end if(std::abs(x[i])<210.215)
+		else if(std::abs(x.back())>=210.215) {
+			x_right_diff.push_back(std::abs(std::abs(x[i]) - 210.25));
+			if(x_right_diff.back()<=right_dist_min) { right_dist_min=x_right_diff.back(); right_dist_min_pos = i; }
+		}//end if(std::abs(x[i])<210.215)
+		xsum+=x.back();
+	}//end loop over track points
+	xavg = xsum/ntrk;
+
+	if(left_dist_min_pos<= (numelmnts-1) && right_dist_min_pos<= (numelmnts-1)) {
+
+		TVector3 leftpt(x[left_dist_min_pos],y[left_dist_min_pos],z[left_dist_min_pos]);
+		TVector3 rightpt(x[right_dist_min_pos],y[right_dist_min_pos],z[right_dist_min_pos]);
+
+		//calculate parameters for a line that passes through the two points:
+		double mxy = (leftpt.Y() - rightpt.Y())/(leftpt.X()-rightpt.X());
+		double mxz = (leftpt.Z() - rightpt.Z())/(leftpt.X()-rightpt.X());
+
+		double bxy = leftpt.Y() - mxy*leftpt.X();
+		double bxz = leftpt.Z() - mxz*leftpt.X();
+
+		if(xavg>0) my_x = 210.215; 
+		else if(xavg<=0) my_x = -210.215;
+		my_y = mxy*my_x + bxy;
+		my_z = mxz*my_x + bxz;
+	}
+
+}//end definition of std::pair<int,int> getCatCrossYZ(std::vector<art::Ptr<recob::Hit>> trk_hits)
+bool icarus::CRTTPCTruthEffMCKalman::willEnterCRTVolume(TVector3 vecpos, TVector3 vecdir, int crtregion){
+
+	bool returnbool;
+
+	struct boundary_plane{
+		double xmin, xmax, ymin, ymax, zmin, zmax;
+		int thin_range;//<range of wall that is thinnest, x=0, y=1, z=2
```

Comment:
There's a variable called "thinrange" (no underscore) declared and used elsewhere. Is "thin_range" different? If not, does it make sense to pass thinrange as a function input here?